File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C/s967693306_commented.c
#include <stdio.h>
#include <string.h>

int main() {
    int required[1001];
    for (int i = 0; i <= 1000; i++) {
        required[i] = -1;
    }
    int n, m;
    scanf("%d %d", &n, &m);
    for (int i = 0; i < m; i++) {
        int s, c;
        scanf("%d %d", &s, &c);
        if (required[s] != -1 && required[s] != c) {
            printf("-1\n");
            return 0;
        }
        required[s] = c;
    }
    for (int num = 0; num <= 999; num++) {
        char s[10];
        sprintf(s, "%d", num);
        int len = strlen(s);
        if (len != n) continue;
        int valid = 1;
        for (int j = 0; j < n; j++) {
            int pos = j + 1;
            int digit = s[j] - '0';
            if (required[pos] != -1) {
                if (digit != required[pos]) {
                    valid = 0;
                    break;
                }
            } else {
                if (digit != 0) {
                    if (j == 0 && digit == 1) {
                        // allowed
                    } else {
                        valid = 0;
                        break;
                    }
                }
            }
        }
        if (valid) {
            printf("%s\n", s);
            return 0;
        }
    }
    printf("-1\n");
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s967693306_commented.py
import sys

def main():
    n, m = map(int, sys.stdin.readline().split())
    constraints = {}
    for _ in range(m):
        s, c = map(int, sys.stdin.readline().split())
        if s in constraints:
            if constraints[s] != c:
                print(-1)
                return
        constraints[s] = c
    for i in range(1000):
        s = str(i)
        if len(s) != n:
            continue
        valid = True
        for j in range(n):
            pos = j + 1
            digit = int(s[j])
            required = constraints.get(pos)
            if required is not None:
                if digit != required:
                    valid = False
                    break
            else:
                if digit != 0:
                    if j != 0 or digit != 1:
                        valid = False
                        break
        if valid:
            print(s)
            return
    print(-1)

if __name__ == "__main__":
    main()
# End of Code
